lib/pacemaker/pcmk_sched_actions.c

```diff
@@ -696,6 +696,56 @@ is_primitive_action(pe_action_t *action)
         }                                                                   \
     } while (0)
 
+/*!
+ * \internal
+ * \brief Update actions in an asymmetric ordering
+ *
+ * \param[in] first  'First' action in an asymmetric ordering
+ * \param[in] then   'Then' action in an asymmetric ordering
+ */
+static void
+handle_asymmetric_ordering(pe_action_t *first, pe_action_t *then)
+{
+    enum rsc_role_e then_rsc_role = RSC_ROLE_UNKNOWN;
+    GList *then_on = NULL;
+
+    if (then->rsc == NULL) {
+        // Asymmetric orderings only matter if there's a resource involved
+        return;
+    }
+
+    then_rsc_role = then->rsc->fns->state(then->rsc, TRUE);
+    then_on = then->rsc->running_on;
+
+    if ((then_rsc_role == RSC_ROLE_STOPPED)
+        && pcmk__str_eq(then->task, RSC_STOP, pcmk__str_none)) {
+        /* Nothing needs to be done for asymmetric ordering if 'then' is
+         * supposed to be stopped after 'first' but is already stopped.
+         */
+        return;
+    }
+
+    if ((then_rsc_role >= RSC_ROLE_STARTED)
+        && pcmk_is_set(then->flags, pe_action_optional)
+        && (then->node != NULL)
+        && pcmk__list_of_1(then_on)
+        && (then->node->details == ((pe_node_t *) then_on->data)->details)
+        && pcmk__str_eq(then->task, RSC_START, pcmk__str_none)) {
```

Comment:
Did you intend to change this comparison and the one above to pcmk__str_none?

Reply:
Yes, I've been doing those as I run across them. Probably should be separate commits but I never remember to do that ...

Reply:
Looks good to me, then.